### ISSUE TITLE
Array mutation pattern with new ```immutable-data```'s ```ignoreNonConstDeclarations``` is not working

### ISSUE THREAD

**Issue by rickmed** [eslint-functional/eslint-plugin-functional#692]
Regarding the feature released as beta in #691 

the rule complains with this:

```typescript
let arr2 = []
for (const x of [1, 2]) {
    arr2.push(x)   // rule complains here with "Modifying an existing object/array is not allowed."
}
```

my .eslintrc:

```json
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint",
    "functional"
  ],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "rules": {
    "semi": ["error", "never"],
    "quotes": ["error", "double", { "allowTemplateLiterals": true }],
    "no-inner-declarations" : "off",
    "no-console": "error",
    "functional/immutable-data": ["error", {
      "ignoreNonConstDeclarations": true
      }],
    "prefer-const": "off"
  },
  "env": {
    "es2022": true,
    "node": true
  }
}
```

installed version is ```eslint-plugin-functional@6.0.0-next.6```

**Comment by rickmed** [comment]
https://github.com/eslint-functional/eslint-plugin-functional/commit/96449949d35b5b8dcbb87f91065f1d08131364fb fix worked for me 👍
